server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -401,7 +401,16 @@ public void onTimeout(TimeValue timeout) {
             }
         }
 
-        injector.getInstance(HttpServerTransport.class).start();
+        var httpTransport = injector.getInstance(HttpServerTransport.class);
+        try {
+            httpTransport.start();
+        } catch (IllegalStateException e) {
+            if (httpTransport.lifecycleState() == Lifecycle.State.CLOSED) {
+                throw new IllegalStateException("Node is shutting down, cannot complete startup", e);
```

Comment:
This should already be handled by `Lifecycle.canMoveToStarted()`

Reply:
I wanted to add a clearer exception message, as this is something which can actually appear to users, so they don't think this is an actual problem

Reply:
Couldn't this theoretically happen with any lifecycle component? We are in the middle of starting up, but we get a SIGTERM and start shutting down. Since they occur in separate threads, any lifecycle component could succumb to this error, not just the http transport.

Can we solve this messaging issue in AbstractLifecycleComponent? eg have a static volatile (perhaps in Elasticsearch or a utility class) that could be checked to determine if we are currently starting up or shutting down. Then when we find an error case in canMoveToStarted() or similar, we can check that and use a more tuned message.

Reply:
Good idea - I've added a helper to determine if the process is shutting down